src/H5Shyper.c

```diff
@@ -12334,7 +12334,7 @@ H5S_hyper_get_first_inc_block(const H5S_t *space, hsize_t clip_size, bool *parti
     if (diminfo->start >= clip_size) {
         ret_value = 0;
         if (partial)
-            partial = false;
+            *partial = false;
```

Comment:
Great catch! Did you happen to see this issue with a tool or just from reading the code?

Reply:
I actually didn't see it before I put `set (C_STANDARD 23)` everywhere. :-)